workflow/controller/operator.go

```diff
@@ -416,6 +416,42 @@ func (woc *wfOperationCtx) processNodeRetries(node *wfv1.NodeStatus, retryStrate
 	return nil
 }
 
+func (woc *wfOperationCtx) collectConditionResults(pod *apiv1.Pod, currentResults *[]wfv1.ErrorResult, annotationKey string) error {
+
+	if resultString, ok := pod.Annotations[annotationKey]; ok {
```

Comment:
this looks so hacky but I get why you did it 😬 

do you understand why the argo team decided that it was a good idea to move data between the wait container and the controller through annotations? 

Any ideas about how this can be done differently?